cmd/rpcfuzz/rpcfuzz.go

```diff
@@ -1656,6 +1676,84 @@ func GetCurrentChainID(ctx context.Context, rpcClient *rpc.Client) (*big.Int, er
 	return chainId, err
 }
 
+func CallRPCAndValidate(ctx context.Context, rpcClient *rpc.Client, currTest RPCTest) testreporter.TestResult {
+	n := 1
+	currTestResult := testreporter.TestResult{
+		Name:             currTest.GetName(),
+		Method:           currTest.GetMethod(),
+		Args:             make([][]interface{}, n),
+		Result:           make([]interface{}, n),
+		Errors:           make([]error, n),
+		NumberOfTestsRan: n,
+	}
+	args := currTest.GetArgs()
+
+	idx := 0 // only one run happening
+	var result interface{}
+	err := rpcClient.CallContext(ctx, &result, currTest.GetMethod(), args...)
+	currTestResult.Args[idx] = args
+	currTestResult.Result[idx] = result
+
+	if err != nil && !currTest.ExpectError() {
+		currTestResult.NumberOfTestsFailed++
+		currTestResult.Errors[idx] = errors.New("Method test failed: " + err.Error())
+		return currTestResult
+	}
+	if err == nil && currTest.ExpectError() {
+		currTestResult.NumberOfTestsFailed++
+		currTestResult.Errors[idx] = errors.New("Expected an error but didn't get one: " + err.Error())
+		return currTestResult
+	}
+
+	if currTest.ExpectError() {
+		err = currTest.Validate(err)
+	} else {
+		err = currTest.Validate(result)
+	}
+
+	if err != nil {
+		currTestResult.NumberOfTestsFailed++
```

Comment:
Same thought with all this `++`. YOu might want something like

```
r := testreporter.New(curTest) // it should be able to get all of the information that it needs from the test interface
r.Passed()
```

Reply:
thanks for the suggestion, i updated it to incorporate that.